spinn_machine/full_wrap_machine.py

```diff
@@ -58,21 +58,21 @@ def get_xy_cores_by_ethernet(self, ethernet_x, ethernet_y):
                 yield(((x + ethernet_x) % self._width,
                       (y + ethernet_y) % self._height), n_cores)
 
-    @overrides(Machine.get_chips_by_ethernet)
-    def get_chips_by_ethernet(self, ethernet_x, ethernet_y):
-        for (x, y) in self._local_xys:
-            chip_xy = ((x + ethernet_x) % self._width,
-                       (y + ethernet_y) % self._height)
-            if (chip_xy) in self._chips:
-                yield self._chips[chip_xy]
-
     @overrides(Machine.get_existing_xys_by_ethernet)
     def get_existing_xys_by_ethernet(self, ethernet_x, ethernet_y):
-        for (x, y) in self._local_xys:
-            chip_xy = ((x + ethernet_x) % self._width,
-                       (y + ethernet_y) % self._height)
-            if (chip_xy) in self._chips:
-                yield chip_xy
+        if self._virtual_chips:
+            for (x, y) in self._local_xys:
+                chip_xy = ((x + ethernet_x) % self._width,
+                           (y + ethernet_y) % self._height)
+                if chip_xy in self._chips and \
+                        chip_xy not in self._virtual_chips:
+                    yield chip_xy
+        else:
+            for (x, y) in self._local_xys:
+                chip_xy = ((x + ethernet_x) % self._width,
+                           (y + ethernet_y) % self._height)
+                if (chip_xy) in self._chips:
+                    yield chip_xy
```

Comment:
Is it worth having this whole loop twice?  Does the check of an empty virtual_chips take that long?

Reply:
It is a check against an empty OrderedDict so no I dont image it very long.

Was speed vs style choice but don't really mind.

Reply:
Probably not worth the split in that case :)